addons/mrp/models/mrp_unbuild.py

```diff
@@ -253,8 +253,10 @@ def _generate_move_from_existing_move(self, move, factor, location_id, location_
         })
 
     def _generate_move_from_bom_line(self, product, product_uom, quantity, bom_line_id=False, byproduct_id=False):
-        location_id = bom_line_id and product.property_stock_production or self.location_id
-        location_dest_id = bom_line_id and self.location_dest_id or product.with_company(self.company_id).property_stock_production
+        product_prod_location = product.with_company(self.company_id).property_stock_production
+        # VFE FIXME shouldn't it be self.location_id or product_prod_location ?
+        location_id = bom_line_id and product_prod_location or self.location_id
+        location_dest_id = bom_line_id and self.location_dest_id or product_prod_location
```

Comment:
okay with the patch
about your comment, not having a bom_line_id means product is a finished produt -> should go from prod to the desired location, so the conditions are right